tools/app/app/components/MetadataView.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { useCardType, useFieldTypes } from '../lib/api'
+import { useTranslation } from 'react-i18next'
+import { Box, Link, Stack } from '@mui/joy'
+import {
+  Controller,
+  FormProvider,
+  useForm,
+  useFormContext,
+} from 'react-hook-form'
+import {
+  CardMetadata,
+  DataType,
+  FieldTypeDefinition,
+  MetadataValue,
+} from '../lib/definitions'
+import { Collapse } from '@mui/material'
+import EditableField from './EditableField'
+
+export interface MetadataViewProps {
+  initialExpanded?: boolean
+  editMode?: boolean
+  metadata?: CardMetadata
+  onClick?: () => void
+}
+
+function MetadataView({
+  initialExpanded,
+  editMode,
+  metadata,
+  onClick,
+}: MetadataViewProps) {
+  const { t } = useTranslation()
+  const [expanded, setExpanded] = useState(initialExpanded)
+
+  const { cardType } = useCardType(metadata?.cardtype ?? null)
+
+  const { fieldTypes } = useFieldTypes()
+
+  const context = useFormContext() // must be inside a <FormProvider>
+  useEffect(() => {
+    setExpanded(initialExpanded ?? false)
+  }, [initialExpanded])
```

Comment:
The prop name "initialExpanded" seems to refer to something that should not change. The view is initially expanded or not and that's it. If it should be used to expand-collapse the view, I think the name should be just "expanded".

Reply:
I removed it